### ISSUE TITLE
Endpoint tries to attach to HNS every time it tries to initialize.

### ISSUE THREAD

**Issue by kayanme** [projectcalico/cni-plugin#988]
Every action, pod net initialization etc. is impossible due to already attach endpoint.

## Expected Behavior
Endpoint should initialize correctly.

## Current Behavior
Now it is always message like this:
2020-12-06 01:48:27.185 [ERROR][1276] dataplane_windows.go 575: Unable to hot attach bridge endpoint [Calico_ep] to host compartment, error: failed during hnsCallRawResponse: hnsCall failed in Win32: This endpoint is already attached to the switch. (0x803b0014) ContainerID="1f291782d93c30ee171607c37d811a5180133b176b8e5cb6d96c89605654caac" Namespace="default" Pod="test-win" WorkloadEndpoint="k8s--windows--1-k8s-test--win-eth0"

## Possible Solution
In dataplane/windows/dataplane_windows.go at 551 change "attachEndpoint = false" to "attachEndpoint := false" to complete endpoint existence check.

## Your Environment

* CNI plugin version 3.17
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes
* CNI config: BGP
* Operating System and version: Windows Server 20H2

**Comment by song-jiang** [comment]
@kayanme Have you deployed any version of Calico for Windows successfully on 20H2? Just wondering if Calico has an compatibility issue with 20H2. I am not sure how easy it is for you to test it on 1903 or 1909.

**Comment by kayanme** [comment]
No, this is my first experiment. :) I can do the same (i.e. attaching to hostnet) on 1809, but I think that this is not Windows core problem here now.  F.e. (do not know Go well) some string case problem when comparing the ids or smth. Other things go fine, BGP peers are creating, routes to the other nodes are processing correctly, the node itself is visible by controller and calicoctl.

**Comment by song-jiang** [comment]
@kayanme Have you done any test on 1809? Do you see the same issue?

**Comment by kayanme** [comment]
Not yet. Actually, I am waiting for some ideas from your side. :) I only tried to repeat the same thing with a newer version of Calico (when it was released) - with the same result.

**Comment by song-jiang** [comment]
@kayanme Are you running on your cluster on any cloud provider? And what is the steps to repo? We could try to repo to take a look. Besides, if you can rerun your test on a Windows 1809 node, that certainly helps to see if it is a compatibility issue.

**Comment by kayanme** [comment]
Yes, it is an Openstack provider, but I am not sure it matters for Calico, because it happens on manually created VM on given virtual network. 
I assembling each part of K8S manually, so it would be hard to describe the whole process.
But in case of Calico it is just:
1. Running setup script with the DownloadOnly mode.
2. Setting up kubelet from that script and needed configs (it is not neccessary to have kubeproxy for the error reproducing).
3. Starting up Calico with my configuration in BGP mode.

Then I can try to create a pod on that node, f.e. Or just initalize kubelet or something. Any thing, that does HNS endpoint check after HNS been created by Calico produces this error.
